internal/daemonset/daemonset.go

```diff
@@ -57,19 +57,12 @@ func (dc *daemonSetGenerator) GarbageCollect(ctx context.Context, mod *kmmv1beta
 	deleted := make([]string, 0)
 
 	for _, ds := range existingDS {
-		if isOlderVersionUnusedDaemonset(&ds, mod.Spec.ModuleLoader.Container.Version) {
-			if err := dc.client.Delete(ctx, &ds); err != nil {
-				return nil, fmt.Errorf("could not delete DaemonSet %s: %v", ds.Name, err)
-			}
+		if isOlderVersionUnusedDaemonset(&ds, mod.Spec.ModuleLoader.Container.Version) ||
+			isModuleLoaderDaemonsetWithInvalidKernel(&ds, dc.kernelLabel, validKernels) {
```

Comment:
This looks unrelated to the rest of the PR.

Reply:
This is just a better refactor of the GarbageCollection

Reply:
added it here, in order not to open a different PR just for that